logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/asset/GetPrivateAssetUseCase.kt

```diff
@@ -0,0 +1,53 @@
+package com.wire.kalium.logic.feature.asset
+
+import com.wire.kalium.cryptography.utils.AES256Key
+import com.wire.kalium.cryptography.utils.EncryptedData
+import com.wire.kalium.cryptography.utils.decryptDataWithAES256
+import com.wire.kalium.logic.CoreFailure
+import com.wire.kalium.logic.data.asset.AssetRepository
+import com.wire.kalium.logic.data.id.ConversationId
+import com.wire.kalium.logic.data.user.UserAssetId
+import com.wire.kalium.logic.functional.suspending
+import com.wire.kalium.logic.kaliumLogger
+
+interface GetPrivateAssetUseCase {
```

Comment:
I don't think the user of Kalium should need to care about if am asset is private or public. I would have expected something like this:

```kotlin
GetMessageAssetUseCase {
    suspend operator fun invoke(
         conversationId: ConversationId,
         messageId: String
     ): AssetResult
}
```

Reply:
I was just following the naming convention that we used for the avatar download. Do you want me to rename both or only this one as mentioned above?

Reply:
I wasn't aware but yeah I would rename both. 

My concern was also about not exposing assetKey, assetToken and encryptionKey to the public. This can all be looked up internally on the asset message.

Reply:
hmmm, I didn't think that would be problematic but sure, the less information we expose the merrier. On it!

Reply:
Pushed requested changes @typfel